### DISCUSSION TITLE
Computed property bug

### DISCUSSION THREAD

**Discussion by nihad-ziyatkhanli** [livewire/livewire#3862] [1. Help]
I have following computed property in livewire component. 
```
    public function getPostsProperty()
    {
        return Post::all();
    }
```
Posts collection is not empty since I have records in database. From view when I call `isset($this->posts)` it returns `false` when it is expected to return `true`. When I call `empty($this->posts)` it returns `true`. You can test this by dumping the result into view.

**Comment by joshhanley** [comment]
@nihad-ziyatkhanli I think that is more of an issue with how PHP's `isset` function works. The computed properties don't actually exist on the component, hence why it's returning false. 

The way the computed properties work is using the `__get` method on the component class. I'd recommend doing some research and seeing if there is a way to detect if the response of a `__get` call.

Hope this helps!

**Reply by nihad-ziyatkhanli** [reply]
> @nihad-ziyatkhanli I think that is more of an issue with how PHP's `isset` function works. The computed properties don't actually exist on the component, hence why it's returning false.
> 
> The way the computed properties work is using the `__get` method on the component class. I'd recommend doing some research and seeing if there is a way to detect if the response of a `__get` call.
> 
> Hope this helps!


That was what I thought. I suspected this had something to do with `isset` not taking result of a function as an argument. However, it didn't explain why `empty` returns opposite of what is expected, since it can take result of a function as an argument. So empty method returns `false` when it is empty, and `true` when it is not. It is the contrary of expectations:

```
dump( empty($this->posts) ); 
/*  `true` when there are records in DB, `false` when there are no records */
```

Isset and empty function behavior generally in PHP:
```
var_dump( empty(array()) ); 
/*  bool(true) */

var_dump( isset(array()) ); 
/* Fatal error: Cannot use isset() on the result of an expression. */
```

**Reply by joshhanley** [reply]
@nihad-ziyatkhanli yeah, who knows 🤷‍♂️ php does have a few quirky things like this at times, where it doesn't exactly make sense.
